### DISCUSSION TITLE
FileNotFoundError: [WinError 2] The system cannot find the file specified

### DISCUSSION THREAD

**Discussion by israfilagir** [ultralytics/ultralytics#6617] [Docs]
My Code:
from ultralytics import YOLO
model = YOLO('yolov8n.pt')  # load a pretrained model (recommended for training)
results = model.train(data=r'C:\Users\ISRAFIL\Desktop\testfolder\Objects365.yaml', epochs=100, imgsz=640)

Objects365.yaml file:
path: C:/Users/Administrator/Desktop/test_training/datasets/Objects365  
train: images/train 
val: images/val
test: 

I get this error when I run my code on the server. Can you please help with the solution?

Windows PowerShell
Copyright (C) 2016 Microsoft Corporation. All rights reserved.

PS C:\Users\Administrator\Desktop\test_training> & C:/Users/Administrator/AppData/Local/Programs/Python/Python310/python.exe c:/Users/Administrator/Desktop/test_trai
ning/test_training_code.py
Ultralytics YOLOv8.0.219 🚀 Python-3.10.0 torch-2.1.1+cpu CPU (Intel Xeon E5-2620 v3 2.40GHz)
engine\trainer: task=detect, mode=train, model=yolov8n.pt, data=C:\Users\Administrator\Desktop\test_training\Objects365.yaml, epochs=30, patience=50, batch=16, imgsz
=640, save=True, save_period=-1, cache=False, device=None, workers=8, project=None, name=train, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0
, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, overlap_mask=Tru
e, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=Non
e, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, show=False, save_frames=False, save_txt=F
alse, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=Fal
se, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.
8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, sc
ale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, cfg=None, tracker=botsort.yaml, save_dir=runs\detect\train

Dataset 'C://Users/Administrator/Desktop/test_training/Objects365.yaml' images not found ⚠️, missing path 'C:\Users\Administrator\Desktop\test_training\datasets\Obje
cts365\images\val'
Processing train in 51 patches ...
Downloading https://dorc.ks3-cn-beijing.ksyun.com/data-set/2020Objects365数据集/train/zhiyuan_objv2_train.tar.gz to 'C:\Users\Administrator\Desktop\test_training\datase
ts\Objects365\zhiyuan_objv2_train.tar.gz'...
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1.24G/1.24G [09:10<00:00, 2.43MB/s]
Unzipping C:\Users\Administrator\Desktop\test_training\datasets\Objects365\zhiyuan_objv2_train.tar.gz to C:\Users\Administrator\Desktop\test_training\datasets\Object
s365...
Traceback (most recent call last):
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python310\lib\site-packages\ultralytics\engine\trainer.py", line 116, in __init__
    self.data = check_det_dataset(self.args.data)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python310\lib\site-packages\ultralytics\data\utils.py", line 321, in check_det_dataset
    exec(s, {'yaml': data})
  File "<string>", line 28, in <module>
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python310\lib\site-packages\ultralytics\utils\downloads.py", line 408, in download
    safe_download(url=u, dir=dir, unzip=unzip, delete=delete, curl=curl, retry=retry)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python310\lib\site-packages\ultralytics\utils\downloads.py", line 333, in safe_download
    subprocess.run(['tar', 'xf' if f.suffix == '.tar' else 'xfz', f, '--directory', unzip_dir], check=True)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python310\lib\subprocess.py", line 501, in run
    with Popen(*popenargs, **kwargs) as process:
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python310\lib\subprocess.py", line 966, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python310\lib\subprocess.py", line 1435, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
FileNotFoundError: [WinError 2] The system cannot find the file specified

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "c:\Users\Administrator\Desktop\test_training\test_training_code.py", line 8, in <module>
    results = model.train(data=r'C:\Users\Administrator\Desktop\test_training\Objects365.yaml', epochs=30, imgsz=640)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python310\lib\site-packages\ultralytics\engine\model.py", line 333, in train
    self.trainer = (trainer or self._smart_load('trainer'))(overrides=args, _callbacks=self.callbacks)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python310\lib\site-packages\ultralytics\engine\trainer.py", line 120, in __init__
    raise RuntimeError(emojis(f"Dataset '{clean_url(self.args.data)}' error ❌ {e}")) from e
RuntimeError: Dataset 'C://Users/Administrator/Desktop/test_training/Objects365.yaml' error  [WinError 2] The system cannot find the file specified

**Comment by pderrenger** [comment]
@israfilagir it seems like the issue you're encountering is related to the unzipping of a dataset file on a Windows system. The error message indicates that the system cannot find the `tar` command, which is typically available on Unix-like systems but not by default on Windows.

To resolve this, you have a couple of options:

1. **Install a Unix-like command-line tool** such as Git Bash, which includes `tar`, or use a package manager like Chocolatey to install `tar` directly on Windows.

2. **Manually unzip the file** using a tool like 7-Zip or WinRAR, and then adjust your dataset path accordingly in your YAML file.

3. **Use Python's built-in libraries** like `tarfile` to programmatically extract the dataset if you're handling the download and extraction within your script.

For now, you might want to manually unzip the dataset and ensure that the paths in your `Objects365.yaml` file are correct and accessible. Make sure that the `train` and `val` directories exist and contain the expected images.

Additionally, ensure that the paths in your YAML file use forward slashes `/` or double backslashes `\\` as path separators, as single backslashes can be interpreted as escape characters in strings.

If you continue to face issues, please check the documentation for further guidance on dataset preparation and training with YOLOv8. 📚
